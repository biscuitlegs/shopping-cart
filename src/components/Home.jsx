import React from 'react'
import { Link } from 'react-router-dom'
import mainImage from '../assets/undraw_shopping_re_hdd9.svg'

const Home = () => (
  <div className="container text-center my-2">
    <h1 className="mb-3 display-1">Welcome to the React Shop!</h1>
    <div className="row gy-5 my-3" data-testid="homeContent">
      <div className="col-lg">
        <img className="img-fluid" src={mainImage} />
      </div>
      <div className="col-lg d-flex align-items-center flex-wrap">
        <h4>
          Ullamco sint eu ea eu ullamco
          <small className="text-muted"> velit nostrud aliquip</small>
        </h4>
        <p className="lead text-start">
          Nostrud proident excepteur aute incididunt in nisi fugiat consectetur
          Lorem minim ut. Est cillum deserunt esse velit consequat irure
          exercitation mollit deserunt reprehenderit laborum labore dolore.
          Nulla dolor est amet occaecat. Esse anim anim ullamco ex proident sit
          eiusmod esse eu fugiat esse officia. Culpa aliquip eu et laborum elit
          magna. Reprehenderit mollit aute mollit ea do ea nulla adipisicing
          consectetur velit cillum pariatur sunt.
        </p>
        <p className="lead text-start">
          Laborum culpa cillum ipsum nisi ut amet proident culpa consequat
          pariatur veniam. Aliquip eu qui eiusmod labore deserunt ex incididunt
          qui labore aute. Anim consectetur aliquip enim excepteur.
        </p>
        <Link className="btn btn-primary w-100" to="/shop">
          Browse Shop
        </Link>
      </div>
    </div>
  </div>
)

export default Home
