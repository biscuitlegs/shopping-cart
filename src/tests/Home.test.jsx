import React from 'react'
import { render, screen, within } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import Home from '../components/Home.jsx'

test('shows a welcome message', () => {
  render(
    <MemoryRouter>
      (<Home />)
    </MemoryRouter>
  )

  const message = screen.getByRole('heading', { name: /Welcome/ })

  expect(message).toBeInTheDocument()
})

test('shows a hero image', () => {
  render(
    <MemoryRouter>
      (<Home />)
    </MemoryRouter>
  )

  const content = screen.getByTestId('homeContent')
  const image = within(content).getByRole('img')

  expect(image).toBeInTheDocument()
})

test('shows sub heading', () => {
  render(
    <MemoryRouter>
      (<Home />)
    </MemoryRouter>
  )

  const content = screen.getByTestId('homeContent')
  const subHeading = within(content).getByRole('heading')

  expect(subHeading).toBeInTheDocument()
})

test('shows paragraphs', () => {
  render(
    <MemoryRouter>
      (<Home />)
    </MemoryRouter>
  )

  const content = screen.getByTestId('homeContent')
  const paragraph1 = within(content).getByText(/Nostrud proident excepteur/)
  const paragraph2 = within(content).getByText(/Laborum culpa cillum/)
  expect(paragraph1).toBeInTheDocument()
  expect(paragraph2).toBeInTheDocument()
})

test('shows a link that opens the shop', () => {
  render(
    <MemoryRouter>
      (<Home />)
    </MemoryRouter>
  )

  const link = screen.getByRole('link')

  expect(link).toBeInTheDocument()
})

test('renders home correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
