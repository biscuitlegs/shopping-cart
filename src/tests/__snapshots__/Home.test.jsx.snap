// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home correctly 1`] = `
<div
  className="container text-center my-2"
>
  <h1
    className="mb-3 display-1"
  >
    Welcome to the React Shop!
  </h1>
  <div
    className="row gy-5 my-3"
    data-testid="homeContent"
  >
    <div
      className="col-lg"
    >
      <img
        className="img-fluid"
        src="undraw_shopping_re_hdd9.svg"
      />
    </div>
    <div
      className="col-lg d-flex align-items-center flex-wrap"
    >
      <h4>
        Ullamco sint eu ea eu ullamco
        <small
          className="text-muted"
        >
           velit nostrud aliquip
        </small>
      </h4>
      <p
        className="lead text-start"
      >
        Nostrud proident excepteur aute incididunt in nisi fugiat consectetur Lorem minim ut. Est cillum deserunt esse velit consequat irure exercitation mollit deserunt reprehenderit laborum labore dolore. Nulla dolor est amet occaecat. Esse anim anim ullamco ex proident sit eiusmod esse eu fugiat esse officia. Culpa aliquip eu et laborum elit magna. Reprehenderit mollit aute mollit ea do ea nulla adipisicing consectetur velit cillum pariatur sunt.
      </p>
      <p
        className="lead text-start"
      >
        Laborum culpa cillum ipsum nisi ut amet proident culpa consequat pariatur veniam. Aliquip eu qui eiusmod labore deserunt ex incididunt qui labore aute. Anim consectetur aliquip enim excepteur.
      </p>
      <a
        className="btn btn-primary w-100"
        href="/shop"
        onClick={[Function]}
      >
        Browse Shop
      </a>
    </div>
  </div>
</div>
`;
